### ISSUE TITLE
Large core.NNN files being created by ArchiveBox docker container

### ISSUE THREAD

**Issue by mfioretti** [pirate/ArchiveBox#278]
Hi,

(writing this as regular issue because I am not sure if it's a bug, or what else...)

I am running the docker image of archivebox, as explained in the archivebox wiki. Works fine... except every now and then it leaves big core files , named "core.NNN". Why does this happen, and what should I do, if anything?

Thanks!

**Comment by pirate** [comment]
Huh, no idea, where are you finding these files on your system?  Are they in the data volume or somewhere else?  It's possible they're the chrome user profile dir, which can get quite big, but is usually never larger than ~4-5gb.

**Comment by mfioretti** [comment]
> Huh, no idea, where are you finding these files on your system? Are they in the data volume or somewhere else? It's possible they're the chrome user profile dir, which can get quite big, but is usually never larger than ~4-5gb.

Here is the info about one such file (414 MBs!) I found by running the container again this morning, on new URLs:

```
#> find . -name "core*" -type f -exec file {} \;
./archive/1571035629/core.56: ELF 64-bit LSB core file x86-64, version 1 (SYSV), SVR4-style, from '/opt/google/chrome-unstable/chrome --type=renderer --no-sandbox --allow-pre-com', real uid: 999, effective uid: 999, real gid: 999, effective gid: 999, platform: 'x86_64'
#> ls -l archive/1571035629/core.56
-rw------- 1 polkitd ssh_keys 414134272 Oct 14 08:48 archive/1571035629/core.56
```

**Comment by pirate** [comment]
Ah it's a core-dump from Chrome crashing, interesting... Can you try running `gdb -c archive/1571035629/core.56` and posting the output here to see what caused the crash?

**Comment by mfioretti** [comment]
> Ah it's a core-dump from Chrome crashing, interesting... Can you try running `gdb -c archive/1571035629/core.56` and posting the output here to see what caused the crash?

No, sorry. There is no gdb on that server. I try to install (=maintain) as little stuff there as possible. My main reason to run archivebox as a container was exactly that.  What I could do is put that core file under the web server directory, just the time needed for you to download it and run the test yourself (but would this make sense on a different computer than the one that generated the core???? ) and then remove it immediately afterwards

**Comment by pirate** [comment]
Downloading it to another machine and running `gdb` there would work fine, it doesn't need to be on the same machine.

**Comment by mfioretti** [comment]
> Downloading it to another machine and running `gdb` there would work fine, it doesn't need to be on the same machine.

sorry, this morning I was distracted and did not think to the most obvious thing to do first, that is download that file on my own machine and running gdb there. I just did it, and this is the result:

```

~$ gdb -c core.57 
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word".
[New LWP 69]
[New LWP 70]
[New LWP 71]
[New LWP 79]
[New LWP 58]
[New LWP 61]
[New LWP 62]
[New LWP 57]
[New LWP 59]
[New LWP 60]
Core was generated by `/opt/google/chrome-unstable/chrome --type=renderer --no-sandbox --allow-pre-com'.
Program terminated with signal SIGILL, Illegal instruction.
#0  0x000055adeee8507d in ?? ()
[Current thread is 1 (LWP 69)]
(gdb) 
```

**Comment by cdvv7788** [comment]
@mfioretti Still having this issue?

**Comment by pirate** [comment]
I cant reproduce this, and many chromium versions have been released since this was first reported, so I think it's safe to close. @mfioretti if you're still seeing it happen on the new `django` branch, comment back here and I can re-open the ticket.

**Comment by pirate** [comment]
Fixed in 8cb5302. It was due to Chrome running out of SHM memory (Docker limits it to 64mb by default, which is unusable).

**Comment by mfioretti** [comment]
> Fixed in [8cb5302](https://github.com/pirate/ArchiveBox/commit/8cb530230c13559e3cbb667ec0e4058cf74355d8). It was due to Chrome running out of SHM memory (Docker limits it to 64mb by default, which is unusable).

THanks! I will download the new container and test this asap.

**Comment by pirate** [comment]
:+1:
```bash
git checkout django
git pull
docker build . -t archivebox
docker run -v $PWD/ouput:/data archivebox init
docker run -v $PWD/ouput:/data archivebox add 'https://example.com'
```
